src/main/java/io/openliberty/tools/intellij/util/DebugModeHandler.java

```diff
@@ -244,40 +246,47 @@ private int getDebuggerTimeoutInSeconds() {
-        String projectPath = libertyModule.getBuildFile().getParent().getPath();
-        Path basePath = null;
-        if (libertyModule.getProjectType().equals(Constants.ProjectType.LIBERTY_MAVEN_PROJECT)) {
-            basePath = Paths.get(projectPath, "target", "liberty", "wlp", "usr", "servers");
+    private Path getServerEnvPath(LibertyModule libertyModule) {
+        String serverDirectory = getServerDirectoryFromLibertyPluginConfig(libertyModule);
+        Path serverEnvPath = Paths.get(serverDirectory, WLP_SERVER_ENV_FILE_NAME);
+        if (Files.exists(serverEnvPath)) {
+            return serverEnvPath;
         } else {
-            basePath = Paths.get(projectPath, "build", "wlp", "usr", "servers");
-        }
-
-        // Make sure the base path exists. If not return null.
-        File basePathFile = new File(basePath.toString());
-        if (!basePathFile.exists()) {
+            LOGGER.trace(String.format("Unable to find the server.env file for project %s", libertyModule.getName()));
             return null;
         }
+    }
 
-        try (Stream<Path> matchedStream = Files.find(basePath, 2, (path, basicFileAttribute) -> {
-            if (basicFileAttribute.isRegularFile()) {
-                return path.getFileName().toString().equalsIgnoreCase(WLP_SERVER_ENV_FILE_NAME);
-            }
-            return false;
-        });) {
-            List<Path> matchedPaths = matchedStream.collect(Collectors.toList());
-            int numberOfFilesFound = matchedPaths.size();
+    /**
+     * Parse the <serverDirectory> value from "liberty-plugin-config.xml" located in target/build directory
+     *
+     * @param libertyModule The Liberty module for which this operations is being performed.
+     *
+     * @return server.env file path
+     *
+     */
+    private String getServerDirectoryFromLibertyPluginConfig(LibertyModule libertyModule) {
+        String serverDirectory = "";
+        try {
+            String projectPath = libertyModule.getBuildFile().getParent().getPath();
+            String projectType = libertyModule.getProjectType().toString();
+            String buildFolder = projectType.equals(Constants.ProjectType.LIBERTY_MAVEN_PROJECT.name()) ? "target" : "build";
+            Path configPath = Paths.get(projectPath, buildFolder, "liberty-plugin-config.xml");
 
-            if (numberOfFilesFound != 1) {
-                if (numberOfFilesFound == 0) {
-                    LOGGER.trace(String.format("Unable to find the server.env file for project %s", libertyModule.getName()));
-                    return null;
-                } else {
-                    throw new Exception(String.format("More than one server.env files were found for project %s. Unable to determine the server.env file to use", libertyModule.getName()));
-                }
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
```

Comment:
Should always use the builtin system default DOM implementation. This matches other XML usage in the plugin. See: https://github.com/OpenLiberty/liberty-tools-intellij/pull/906

```suggestion
            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newDefaultInstance();
```

Reply:
Modified. Thank you @mrglavas